app/assets/javascripts/training/components/training_slide_handler.jsx

```diff
@@ -118,7 +123,7 @@ const TrainingSlideHandler = () => {
     document.title = `${slideTitle} - ${baseTitle}`;
 
     return () => {
-      // cleanup
+      // cleanup. Removes the old event listener
```

Comment:
Does this cleanup run every time the reduce store changes, or only when the component unmounts?

Reply:
Everytime, since the needs the updated value of `training`. When the component unmounts it removes the event listener for the last time.